### PR TITLE
Features/error report improvements

### PR DESCRIPTION
**Pull request by MikeNeilson**
I'm probably missing most of the places where this should be added but the PR is already getting a bit large.

Primarily moved to a mechanism of controllers and daos and such to be able to say, "yeah this is bad report to user, here's the info" and let a more central place decide just how much to report.

For example a required property can be configured with an annotation and the json library error caught and adjusted.

But a mechanism for more complex logic, like a mutually exclusive set of fields was allow added so that the exact program can be reported to the user vs them having to ask for help and give an incident identifier for well, rather basic issues.

### CODE REVIEW

File: .vscode/launch.json [hunk 1]
```diff
@@ -26,22 +26,33 @@
                 "RADAR_DEBUG_LOGGING": "true"
             },
             "sourcePaths": [
-                "build/generated/sources/annotationProcessor/java/main"
+                "**src/main/**",
+                "build/generated/sources/annotationProcessor/java/main",
+                "/home/mike/projects/javalin/**src/main/java/**"
```
Comment (rripken) [inline .vscode/launch.json]: Its a little weird and not portable that /home/mike paths have crept into the repo but I haven't been using vscode so idk.  If its helpful to keep the file in the repo I guess its fine.

File: cwms_radar_api/src/main/java/cwms/radar/ApiServlet.java [hunk 2]
```diff
@@ -239,7 +255,7 @@ public static void radarCrud(@NotNull String path, @NotNull CrudHandler crudHand
             throw new IllegalArgumentException("CrudHandler requires a resource base at the beginning of the provided path, e.g. '/users/{user-id}'");
         }
 
-        Map<CrudFunction, Handler> crudFunctions = getHanders(crudHandler, resourceId);
+        Map<CrudFunction, Handler> crudFunctions = CrudHandlerKt.getCrudFunctions(crudHandler, resourceId);//getHanders(crudHandler, resourceId);
```
Comment (rripken) [inline cwms_radar_api/src/main/java/cwms/radar/ApiServlet.java:258]: I'm glad you got this working.  I could have sworn I tried to call the function this exact way.

**Review by rripken — DISMISSED** [review]
Its a little weird and not portable that /home/mike paths have crept into the repo but I haven't been using vscode so idk.  If its helpful to keep the file in the repo I guess its fine.

File: .vscode/launch.json (hunk 1 above)
Comment (MikeNeilson) [reply .vscode/launch.json]: Oops. That was me trying to look at something and those don't need to be there.

**Review by rripken — APPROVED** [review]

**Pull request merged**